src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs

```diff
@@ -229,14 +229,31 @@ protected static string RunProcess (string exe, string args)
 				CreateNoWindow = true,
 				WindowStyle = ProcessWindowStyle.Hidden,
 			};
-			using (var proc = Process.Start (info)) {
+			using (var proc = new Process ()) {
+				StringBuilder standardOutput = new StringBuilder (), errorOutput = new StringBuilder ();
+				proc.StartInfo = info;
+				proc.OutputDataReceived += new DataReceivedEventHandler ((sender, e) => {
+					if (!string.IsNullOrEmpty (e.Data))
+						standardOutput.Append (Environment.NewLine + e.Data);
+				});
+				proc.ErrorDataReceived += new DataReceivedEventHandler ((sender, e) => {
+					if (!string.IsNullOrEmpty (e.Data))
+						errorOutput.Append (Environment.NewLine + e.Data);
+				});
+
+				proc.Start ();
+				proc.BeginOutputReadLine ();
+				proc.BeginErrorReadLine ();
+
 				if (!proc.WaitForExit ((int)TimeSpan.FromSeconds (30).TotalMilliseconds)) {
 					proc.Kill ();
 					TestContext.Out.WriteLine ($"{nameof (RunProcess)} timed out: {exe} {args}");
 					return null; //Don't try to read stdout/stderr
 				}
-				var result = proc.StandardOutput.ReadToEnd ().Trim () + proc.StandardError.ReadToEnd ().Trim ();
-				return result;
+
+				proc.WaitForExit ();
```

Comment:
Do we need the second `WaitForExit()`? If it got here, the first one returned `true`?

Reply:
I agree awe shouldn't need this.

Reply:
Actually we do, from the documentation for WaitForExit(int):
_When standard output has been redirected to asynchronous event handlers, it is possible that output processing will not have completed when this method returns. To ensure that asynchronous event handling has been completed, call the WaitForExit() overload that takes no parameter after receiving a true from this overload._ (https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=netframework-4.7.2#System_Diagnostics_Process_WaitForExit_System_Int32_)